### PR TITLE
Big rock

### PR DESCRIPTION
**Pull request by Tokoeka**
get Caveman Dans Big Rocks for your collection - uses pro skatboard epic mctwist and goose drones to dupe if possible

### CODE REVIEW

File: src/rocks.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        get("_questCaveDan", 0) > 5 ||
+        get("lastCaveDanDefeat", 0) >= myAscensions(),
+      do: () => {
+        visitUrl("place.php?whichplace=twitch&action=twitch_dancave3");
+        set("_questCaveDan", 6);
+        set("lastCaveDanDefeat", myAscensions());
+      },
+      outfit: () => {
+        return chooseQuestOutfit(
+          { location, isFree: getKramcoWandererChance() >= 1 },
```
Comment (Rinn) [inline src/rocks.ts:183]: can remove the isFree kramco chance

File: src/rocks.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  set,
+} from "libram";
+
+const location = $location`The Cave Before Time`;
+export const bigRock: ChronerQuest = {
+  name: "Get Rock",
+  location,
+  tasks: [
+    {
+      name: "getHit",
```
Comment (horrible-little-slime) [inline src/rocks.ts:29]: I think a lot of these tasks should be renamed--it's outward-facing content that's worth having a clear, descriptive name

File: src/rocks.ts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        visitUrl(toUrl(location));
+        for (const choiceValue of [3, 1, 2, 1, 2]) {
+          runChoice(choiceValue);
+        }
+        /* if (handlingChoice()) {
+          runChoice(1); // 954 - Ask About Password
+          runChoice(2); // 954 - Game of Chance
+          runChoice(1); // 954 - Pick Rock
+          runChoice(2); // 954 - No Thanks
+        } */
```
Comment (horrible-little-slime) [inline src/rocks.ts:55]:
```suggestion
```

File: src/rocks.ts [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        get("_questCaveDan", 0) > 3 ||
+        get("lastCaveDanDefeat", 0) >= myAscensions(),
+      do: () => {
+        visitUrl(toUrl(location));
+        if (handlingChoice()) {
+          runChoice(3); // 955 - Go towards noise
+          runChoice(1); // 954 - Ask About Password
+          runChoice(1); // 954 - Offer to Trade
+          runChoice(2); // 954 - Teach Secret of Scissors
+        }
```
Comment (horrible-little-slime) [inline src/rocks.ts:119]:
```suggestion
        for (const choiceValue of [3, 1, 1, 2]) {
          runChoice(choiceValue);
        }
```

File: src/setup.ts [hunk 5]
```diff
@@ -135,7 +135,7 @@ export const setup: Quest<ChronerTask> = {
           $locations`The Post-Mall, Moonshiners' Woods, The Cave Before Time, The Sleazy Back Alley`,
         );
       },
-      ready: () => AutumnAton.available(),
+      ready: () => AutumnAton.available() && !get("noncombatForcerActive"),
```
Comment (horrible-little-slime) [inline src/setup.ts:138]: I don't see why this is a necessary condition for that.

File: src/outfit.ts [hunk 6]
```diff
@@ -48,7 +47,10 @@ export function chooseQuestOutfit(
   { location, isFree }: TaskOptions,
   ...outfits: OutfitSpec[]
 ): OutfitSpec {
-  const familiar = chooseFamiliar({ location });
+  const mergedInputSpec = mergeSpecs(...outfits);
+  const familiar = mergedInputSpec.familiar
+    ? mergedInputSpec.familiar
+    : chooseFamiliar({ location });
```
Comment (horrible-little-slime) [inline src/outfit.ts:53]:
```suggestion
  const familiar = mergedInputSpec.familiar ?? chooseFamiliar({ location });
```

File: src/setup.ts (hunk 5 above)
Comment (Tokoeka) [reply src/setup.ts:138]: Hmm, you might be right  - looking back over my logs, I think I misunderstood which action was tripping it up. will remove

**Review by Rinn — APPROVED** [review]

**Pull request merged**